From 2b8022764a1a89df0d60289a2acec0a649c46efe Mon Sep 17 00:00:00 2001
From: Mark Charlebois <charlebm@gmail.com>
Date: Wed, 12 Feb 2014 12:16:58 -0800
Subject: [PATCH] arm64: LLVMLinux: Check for NULL in
 futext_atomic_cmpxchg_inatomic()

futex_init passes in uaddr = 0. The LLVM optimizer will see that *uaddr is 0, set the register
to xzr and compilation will fail.

Signed-off-by: Mark Charlebois <charlebm@gmail.com>
---
 arch/arm64/include/asm/futex.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- linux.orig/arch/arm64/include/asm/futex.h
+++ linux/arch/arm64/include/asm/futex.h
@@ -108,7 +108,7 @@
 	int ret = 0;
 	u32 val, tmp;
 
-	if (!access_ok(VERIFY_WRITE, uaddr, sizeof(u32)))
+	if (uaddr == 0 || !access_ok(VERIFY_WRITE, uaddr, sizeof(u32)))
 		return -EFAULT;
 
 	asm volatile("// futex_atomic_cmpxchg_inatomic\n"
